tools/tpm2_pkcs11/db.py

```diff
@@ -252,6 +255,7 @@ def create(self):
                 objauth TEXT NOT NULL,
                 mech TEXT NOT NULL,
                 attrs TEXT NOT NULL,
+                tobjauthpolicy BLOB,
```

Comment:
how can objauth remain non-null in the case of policy hash? We're storing the policy hash here right? If it's the same hash, why duplicate it throughout the whole db?

Reply:
You are right. It cannot be a Null. I was phasing the changes in commits so I don't break the existing setup. So in the new auth model the concept of tobjects will vanish. In general though, that policypassword is a placeholder to bring other changes like policy based authorizations into the pkcs11 library. Once there, in the next phases/ commits the policy for the seal objects would be different from the primary objects.

Reply:
> You are right. It cannot be a Null. I was phasing the changes in commits so I don't break the existing setup. So in the new auth model the concept of tobjects will vanish. 

Why would the concept of tobjects vanish, they are the actual backing keys and objects the user interacts with via the pkcs11 API. Why would they go away?

> In general though, that policypassword is a placeholder to bring other changes like policy based authorizations into the pkcs11 library. Once there, in the next phases/ commits the policy for the seal objects would be different from the primary objects.

Their's two policies IIUC.  A policy for interacting with the "seal objects" and a policy for accessing the "tobjects". The soseal object would have 1 policy hash, the userseal object would have 1 policy hash and all the tobjects that use policy would share common hash.

Reply:
Anything that is shared in common in a db should be placed in a separate table with the value looked up by primary id. It helps prevent the db from bloating and normalizing databases.

Reply:
> > You are right. It cannot be a Null. I was phasing the changes in commits so I don't break the existing setup. So in the new auth model the concept of tobjects will vanish.
> 
> Why would the concept of tobjects vanish, they are the actual backing keys and objects the user interacts with via the pkcs11 API. Why would they go away?
> 
> > In general though, that policypassword is a placeholder to bring other changes like policy based authorizations into the pkcs11 library. Once there, in the next phases/ commits the policy for the seal objects would be different from the primary objects.
> 
> Their's two policies IIUC. A policy for interacting with the "seal objects" and a policy for accessing the "tobjects". The soseal object would have 1 policy hash, the userseal object would have 1 policy hash and all the tobjects that use policy would share common hash.

That was my confusion. By tobjects I thought the parent objects were not primary objects which is why I thought that should vanish.